sched/adaptdl_sched/policy/pollux.py

```diff
@@ -310,14 +349,25 @@ def _mutation(self, states, **kwargs):
         prob = 1.0 / np.where(states > 0, num_nonzero, num_zero)
         prob = prob.reshape(states.shape)
         m = np.random.random(states.shape) < prob
-        r = np.random.randint(np.iinfo(np.int16).max, size=states.shape)
+        r = np.random.randint(self._min_replicas, self._max_replicas + 1,
+                              size=states.shape)
         states = (states + m * r) % (self._max_replicas + 1)
+        # We need at least min_replicas for jobs with min_replicas > 0
+        min_replicas_idx = [i for i, j in enumerate(self._jobs)
+                            if j.min_replicas > 0]
+        states[:, min_replicas_idx] = np.maximum(
+                                          states[:, min_replicas_idx],
+                                          self._min_replicas[min_replicas_idx])
```

Comment:
Are lines 356-360 just the same as `states = np.maximum(states, self._min_replicas)`?

Reply:
yes. simplified the assignment.